vlm/src/main/scala/geotrellis/contrib/vlm/spark/RasterSourceRDD.scala

```diff
@@ -128,7 +143,15 @@ object RasterSourceRDD {
     readingSourcesRDD: RDD[ReadingSource],
     layout: LayoutDefinition,
     partitioner: Option[Partitioner]
-  )(implicit sc: SparkContext): MultibandTileLayerRDD[SpatialKey] = {
+  )(implicit sc: SparkContext): MultibandTileLayerRDD[SpatialKey] =
+    readK[SpatialKey](readingSourcesRDD, layout, (_, sk) => sk, partitioner)
+
+  def readK[K: SpatialComponent: TileToLayoutRasterSource: Ordering: ClassTag](
+    readingSourcesRDD: RDD[ReadingSource],
+    layout: LayoutDefinition,
+    keyTransformation: (RasterSource, SpatialKey) => K,
```

Comment:
In fact the entire solution is in a smart function that can lift `SpatialKey` to any arbitrary `K` using the `RasterSource` (and `RasterSourceMetadata` if we'll rebase it on top of https://github.com/geotrellis/geotrellis-contrib/pull/216)